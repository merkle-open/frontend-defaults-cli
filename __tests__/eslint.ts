/// <reference types="@types/jest" />
import { apiIt } from './shared';

describe('eslint', () => {
	it('default', async () =>
		await apiIt('eslint', {
			eslint: true,
		}));
	it('with-prettier', async () =>
		await apiIt('eslint-with-prettier', {
			eslint: true,
			prettier: true,
		}));
	it('with-githooks', async () =>
		await apiIt('eslint-with-githooks', {
			eslint: true,
			githooks: true,
		}));
});
