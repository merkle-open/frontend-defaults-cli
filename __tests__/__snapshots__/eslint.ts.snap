// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint default 1`] = `
"/.idea/
/node_modules/
/*.js
/dist
/build
/coverage
"
`;

exports[`eslint default 2`] = `
"module.exports = {
	extends: [
		'./.eslintrc.strict.js',
	].map(require.resolve),
	/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 */
	rules: {
		'no-console': 1
	}
};
"
`;

exports[`eslint default 3`] = `
"module.exports = {
	extends: ['@namics/eslint-config/configurations/es8-react.js'].map(require.resolve),
};
"
`;

exports[`eslint default 4`] = `
"{
  \\"options\\": {
    \\"ts\\": false,
    \\"es\\": false,
    \\"eslint\\": true,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": false,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": false,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"build\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:js\\": \\"eslint . -c .eslintrc.strict.js\\"
        },
        \\"devDependencies\\": {
          \\"@namics/eslint-config\\": \\"8.0.0\\",
          \\"eslint\\": \\"6.2.1\\",
          \\"eslint-plugin-import\\": \\"2.18.2\\",
          \\"npm-run-all\\": \\"4.1.5\\"
        }
      }
    },
    \\".eslintrc.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = {\\\\n\\\\textends: [\\\\n\\\\t\\\\t'./.eslintrc.strict.js',\\\\n\\\\t].map(require.resolve),\\\\n\\\\t/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 */\\\\n\\\\trules: {\\\\n\\\\t\\\\t'no-console': 1\\\\n\\\\t}\\\\n};\\\\n\\"
    },
    \\".eslintrc.strict.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = {\\\\n\\\\textends: ['@namics/eslint-config/configurations/es8-react.js'].map(require.resolve),\\\\n};\\\\n\\"
    },
    \\".eslintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"/.idea/\\\\n/node_modules/\\\\n/*.js\\\\n/dist\\\\n/build\\\\n/coverage\\\\n\\"
    }
  }
}"
`;

exports[`eslint default 5`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:js\\": \\"eslint . -c .eslintrc.strict.js\\"
  },
  \\"devDependencies\\": {
    \\"@namics/eslint-config\\": \\"8.0.0\\",
    \\"eslint\\": \\"6.2.1\\",
    \\"eslint-plugin-import\\": \\"2.18.2\\",
    \\"npm-run-all\\": \\"4.1.5\\"
  }
}"
`;

exports[`eslint default 6`] = `
Array [
  .eslintignore,
  .eslintrc.js,
  .eslintrc.strict.js,
  .frontend-defaults-rc.json,
  package.json,
]
`;

exports[`eslint with-githooks 1`] = `
/.idea/
/node_modules/
/*.js
/dist
/build
/coverage

`;

exports[`eslint with-githooks 2`] = `
module.exports = {
	extends: [
		'./.eslintrc.strict.js',
	].map(require.resolve),
	/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 */
	rules: {
		'no-console': 1
	}
};

`;

exports[`eslint with-githooks 3`] = `
module.exports = {
	extends: ['@namics/eslint-config/configurations/es8-react.js'].map(require.resolve),
};

`;

exports[`eslint with-githooks 4`] = `
{
  "options": {
    "ts": false,
    "es": false,
    "eslint": true,
    "editorconfig": false,
    "prettier": false,
    "stylelint": false,
    "gitignore": false,
    "npmrc": false,
    "readme": false,
    "githooks": true,
    "commitlint": false,
    "nodeVersion": false,
    "webpack": false,
    "build": false,
    "install": false,
    "force": true,
    "dryRun": false,
    "mode": "api"
  },
  "mergedFiles": {
    "package.json": {
      "override": false,
      "data": {
        "scripts": {
          "lint": "npm-run-all lint:*",
          "lint:js": "eslint . -c .eslintrc.strict.js"
        },
        "devDependencies": {
          "@namics/eslint-config": "8.0.0",
          "eslint": "6.2.1",
          "eslint-plugin-import": "2.18.2",
          "husky": "3.0.4",
          "lint-staged": "9.2.3",
          "npm-run-all": "4.1.5"
        },
        "husky": {
          "hooks": {
            "pre-commit": "lint-staged"
          }
        },
        "lint-staged": {
          "src/**/*.{js,jsx}": [
            "eslint"
          ]
        }
      }
    },
    ".huskyrc": {
      "override": false,
      "data": "# ~/.huskyrc/n# Note: it will run in /bin/sh shell/nif command -v nodenv >/dev/null 2>&1; then/n  eval /"$(nodenv init -)/"/nfi"
    },
    ".eslintrc.js": {
      "override": false,
      "data": "module.exports = {/n/textends: [/n/t/t'./.eslintrc.strict.js',/n/t].map(require.resolve),/n/t/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 *//n/trules: {/n/t/t'no-console': 1/n/t}/n};/n"
    },
    ".eslintrc.strict.js": {
      "override": false,
      "data": "module.exports = {/n/textends: ['@namics/eslint-config/configurations/es8-react.js'].map(require.resolve),/n};/n"
    },
    ".eslintignore": {
      "override": false,
      "data": "/.idea//n/node_modules//n/*.js/n/dist/n/build/n/coverage/n"
    }
  }
}
`;

exports[`eslint with-githooks 5`] = `
# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval "$(nodenv init -)"
fi
`;

exports[`eslint with-githooks 6`] = `
{
  "scripts": {
    "lint": "npm-run-all lint:*",
    "lint:js": "eslint . -c .eslintrc.strict.js"
  },
  "devDependencies": {
    "@namics/eslint-config": "8.0.0",
    "eslint": "6.2.1",
    "eslint-plugin-import": "2.18.2",
    "husky": "3.0.4",
    "lint-staged": "9.2.3",
    "npm-run-all": "4.1.5"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "src/**/*.{js,jsx}": [
      "eslint"
    ]
  }
}
`;

exports[`eslint with-githooks 7`] = `
Array [
  .eslintignore,
  .eslintrc.js,
  .eslintrc.strict.js,
  .frontend-defaults-rc.json,
  .huskyrc,
  package.json,
]
`;

exports[`eslint with-prettier 1`] = `
/.idea/
/node_modules/
/*.js
/dist
/build
/coverage

`;

exports[`eslint with-prettier 2`] = `
module.exports = {
	extends: [
		'./.eslintrc.strict.js',
	].map(require.resolve),
	/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 */
	rules: {
		'no-console': 1
	}
};

`;

exports[`eslint with-prettier 3`] = `
module.exports = {
	extends: [
		'@namics/eslint-config/configurations/es8-react.js',
		'@namics/eslint-config/configurations/es8-react-disable-styles.js',
	].map(require.resolve),
};

`;

exports[`eslint with-prettier 4`] = `
{
  "options": {
    "ts": false,
    "es": false,
    "eslint": true,
    "editorconfig": false,
    "prettier": true,
    "stylelint": false,
    "gitignore": false,
    "npmrc": false,
    "readme": false,
    "githooks": false,
    "commitlint": false,
    "nodeVersion": false,
    "webpack": false,
    "build": false,
    "install": false,
    "force": true,
    "dryRun": false,
    "mode": "api"
  },
  "mergedFiles": {
    "package.json": {
      "override": false,
      "data": {
        "scripts": {
          "lint": "npm-run-all lint:*",
          "lint:js": "eslint . -c .eslintrc.strict.js",
          "prettier": "prettier --write /"**/*.*(js|jsx|ts|tsx|json)/""
        },
        "devDependencies": {
          "@namics/eslint-config": "8.0.0",
          "@namics/prettier-config": "0.2.0",
          "eslint": "6.2.1",
          "eslint-plugin-import": "2.18.2",
          "npm-run-all": "4.1.5",
          "prettier": "1.18.2"
        }
      }
    },
    ".prettierignore": {
      "override": false,
      "data": "# Node/n**/node_modules/n**/package.json/n**/package-lock.json/n/n# Generated/n**/bower.json/n**/lerna.json/n/n# Build/n**/build/n**/dist/n**/public/n**/coverage/n**/storybook-static/n/n# Nitro/n**/project/blueprints/n**/.yo-rc.json"
    },
    ".prettierrc.js": {
      "override": false,
      "data": "module.exports = require('@namics/prettier-config');/n"
    },
    ".eslintrc.js": {
      "override": false,
      "data": "module.exports = {/n/textends: [/n/t/t'./.eslintrc.strict.js',/n/t].map(require.resolve),/n/t/* define rules for IDE only here https://github.com/namics/frontend-defaults/issues/12 *//n/trules: {/n/t/t'no-console': 1/n/t}/n};/n"
    },
    ".eslintrc.strict.js": {
      "override": false,
      "data": "module.exports = {/n/textends: [/n/t/t'@namics/eslint-config/configurations/es8-react.js',/n/t/t'@namics/eslint-config/configurations/es8-react-disable-styles.js',/n/t].map(require.resolve),/n};/n"
    },
    ".eslintignore": {
      "override": false,
      "data": "/.idea//n/node_modules//n/*.js/n/dist/n/build/n/coverage/n"
    }
  }
}
`;

exports[`eslint with-prettier 5`] = `
# Node
**/node_modules
**/package.json
**/package-lock.json

# Generated
**/bower.json
**/lerna.json

# Build
**/build
**/dist
**/public
**/coverage
**/storybook-static

# Nitro
**/project/blueprints
**/.yo-rc.json
`;

exports[`eslint with-prettier 6`] = `
module.exports = require('@namics/prettier-config');

`;

exports[`eslint with-prettier 7`] = `
{
  "scripts": {
    "lint": "npm-run-all lint:*",
    "lint:js": "eslint . -c .eslintrc.strict.js",
    "prettier": "prettier --write /"**/*.*(js|jsx|ts|tsx|json)/""
  },
  "devDependencies": {
    "@namics/eslint-config": "8.0.0",
    "@namics/prettier-config": "0.2.0",
    "eslint": "6.2.1",
    "eslint-plugin-import": "2.18.2",
    "npm-run-all": "4.1.5",
    "prettier": "1.18.2"
  }
}
`;

exports[`eslint with-prettier 8`] = `
Array [
  .eslintignore,
  .eslintrc.js,
  .eslintrc.strict.js,
  .frontend-defaults-rc.json,
  .prettierignore,
  .prettierrc.js,
  package.json,
]
`;
